Assets/_Scripts/1Data/Scriptables/BuildingType.cs

```diff
@@ -1,43 +1,35 @@
+using OrcaDefense.Models;
 using System;
 using System.Collections.Generic;
 using UnityEngine;
 
 [CreateAssetMenu(menuName = "Building Type")]
 public class BuildingType : ScriptableObject
 {
-    public BuildingType()
-    {
-        BuildingValidator = GetValidatorByBuildingType(BuildingEnum);
-    }
-
-    public BuildingType Construct()
-    {
-        return new BuildingType();
-    }
-
     public GameObject Prefab;
 
     public List<ResourceValue> Cost;
 
     public bool hasRoof;
 
-    private BuildingTypeEnum BuildingTypeEnum;
-
-    public BuildingTypeEnum BuildingEnum { get => BuildingTypeEnum; }
+    public BuildingTypeEnum BuildingEnum;
```

Comment:
Can we rename this to BuildingType?

Reply:
Can't be named the same thing as the class. Feel free to rename the class to make this possible though